File: GraalSamples/objectexample.py

import polyglot

class TestSubject:
    
    def __init__(self, age=42, name="John", scientist=True):
        self.age = age
        self.name = name
        self.scientist = scientist
        
    def foo(self):
        return len(self.name)
    
    def __str__(self):
        if self.scientist:
            return self.name + f" is a {self.age} years old scientist"
        return self.name + f" is a {self.age} years old test subject"
    
test = TestSubject()

print("Displaying John in python : ")
print(test)

polyglot.export_value(value=test, name='John')

File: GraalSamples/objectexample.js
Polyglot.evalFile('python', 'objectexample.py');

var john = Polyglot.import('John');
var test = john.foo();

console.log("Displaying John in JS :");
console.log(john);
console.log(john.__str__());

john.age = 43;

console.log("Displaying John in python again after changing a field in JS : ");

Polyglot.eval('python', 'print(test)');